src/Jackett.Common/Definitions/matrix.yml

```
@@ -0,0 +1,184 @@
+    - {id: 47, cat: Other, desc: "Klippek"}
+    - {id: 30, cat: Audio/Lossless, desc: "Lossless/EN"}
+    - {id: 29, cat: Audio/Lossless, desc: "Lossless/HU"}
+    - {id: 25, cat: Audio/MP3, desc: "MP3/EN"}
+    - {id: 6, cat: Audio/MP3, desc: "MP3/HU"}
+    - {id: 33, cat: PC, desc: "Program/egyéb"}
+    - {id: 1, cat: PC/ISO, desc: "Program/ISO"}
+    - {id: 36, cat: TV/HD, desc: "Sorozat/EN/HD"}
+    - {id: 49, cat: TV/SD, desc: "Sorozat/EN/SD"}
+    - {id: 28, cat: TV/HD, desc: "Sorozat/HU/HD"}
+    - {id: 7, cat: TV/SD, desc: "Sorozat/HU/SD"}
+    - {id: 9, cat: XXX, desc: "XXX"}
+    - {id: 44, cat: XXX/x264, desc: "XXX/HD"}
+    - {id: 43, cat: XXX/SD, desc: "XXX/SD"}
+
+  modes:
+    search: [q]
+    tv-search: [q, season, ep, imdbid]
+    movie-search: [q, imdbid]
+    book-search: [q]
+    music-search: [q]
+
+settings:
+  - name: username
+    type: text
+    label: Username
+  - name: password
+    type: password
+    label: Password
+  - name: freeleech
+    type: checkbox
+    label: Filter freeleech only
+    default: false
+  - name: sort
+    type: select
+    label: Sort requested from site
+    default: 0
+    options:
+      0: created
+      1: title
+      4: size
+      6: seeders
+  - name: type
+    type: select
+    label: Order requested from site
+    default: desc
+    options:
+      desc: desc
+      asc: asc
+  - name: info_tpp
+    type: info
+    label: Results Per Page
+    default: For best results, change the <b>Torrents per page:</b> setting to <b>100</b> on your account profile.
+  - name: info_activity
+    type: info
+    label: Account Inactivity
+    default: "If you do not log in to the site for 6 months, the system will automatically delete it!"
+
+login:
+  path: takelogin.php
+  method: post
+  inputs:
+    username: "{{ .Config.username }}"
+    password: "{{ .Config.password }}"
+    logout: ""
+  error:
+    - selector: td.embedded:contains("Sikertelen bejelentkezés")
+  test:
+    path: /
+    selector: a[href$="logout.php"]
+
+search:
+  paths:
+    - path: browse.php
+  inputs:
+    $raw: "{{ range .Categories }}c{{.}}=1&{{end}}"
+    search: "{{ if .Query.IMDBID }}{{ .Query.IMDBID }}{{ else }}{{ .Keywords }}{{ end }}"
+    # 0 active only, 1 include dead, 2 only dead
+    incldead: 1
+    sort: "{{ .Config.sort }}"
+    type: "{{ .Config.type }}"
+
+  rows:
+    selector: "table.mainouter > tbody > tr > td > table tr:has(a[href^=\"download.php?id=\"]){{ if .Config.freeleech }}:has(img[src$=\"/pic/ingyentorrent.gif\"]){{ else }}{{ end }}"
+
+  fields:
+    category:
+      selector: a[href^="browse.php?cat="]
+      attribute: href
+      filters:
+        - name: querystring
+          args: cat
+    title:
+      selector: a.torrent-link b
+    _id:
+      selector: a[href^="download.php?id="]
+      attribute: href
+      filters:
+        - name: querystring
+          args: id
+    details:
+      text: "details.php?id={{ .Result._id }}"
```

```suggestion
    details:
      selector: a[href^="details.php?id="]
      attribute: href
      filters:
        - name: re_replace
          args: ["&hit=1.+", ""]
```

Probably the tidiest version for this, no need for `_id`.

Everything else is good to go!

---

That was a change that @garfield69 requested. Which of you wins? ;-)

---

[Context](https://github.com/Jackett/Jackett/pull/15887#discussion_r1980245381)

---

You were prepending `details.php?id=` despite using a selector with `details.php?id=`, and then using `querystring` to add the `id`. It's a roundabout way of removing the `&hit=1...`, which can be done with a re_replace instead. You can use mine above.

---

Hmm, I think extracting the data we need explicitly and then assembling the URL from that data is more robust if the HTML changes in the future. What if the `hit=` parameter changes? What if a parameter is added before it? It's better to let the, I assume standards compliant, query string parser give us what we want regardless of how the parameters are ordered or what other parameters are included. Then explicitly assemble the needed URL. If the *correct* details URL weren't burried in the results of the JS execution, then I'd be all for using that instead, but in this case I think explicit is less fragile and more maintainable.

LMK if you insist.

---

Equally, what if `id` changes to `ids`? We deal with changes on a regular basis, likewise we replace `hit=1` often enough - https://github.com/search?q=repo%3AJackett%2FJackett+replace+hit+language%3AYAML&type=code&l=YAML

---

> Equally, what if `id` changes to `ids`?

Yes, if the parameter we *want* changes, then yes, there's no avoiding the need for a change. But for all *other* parameter changes, a standards compliant parser is much less fragile. No need to make all those other cases more fragile just for the less likely case that requires changes.

---

We use the `details` link if available, and if minor replacements or removals are needed then we do that as well. That's the same for the `downloads` link or any other. Manually constructing URLs is a last resort.

As much as possible, we try to replicate what a user would do on the site when it comes to searches and links clicked (working around redirects or actual site issues when needed), as we don't want an obvious 'Jackett footprint'. There have been exceptions to this when working site devs, where we intentionally use an input or header specific to Jackett, as it makes their lives easier.

If something were to change with the details link for this site in future, either  it just works because we're just trimming the problematic bits, or it's more obvious in errors, because details.php isn't found. Also, if the downloads.php link changes (e.g. downloads2.php or downloads_ssl.php), then it's one less thing for us to change.

---

> we don't want an obvious 'Jackett footprint'

Ah, that's a factor I hadn't considered. Unfortunately, it doesn't work in favor of either argument over the other. If the details URL the web UI directs users to changes the query parameters, then the logs for my approach will only show valid URLs that the web UI wouldn't generate, IOW you'd have to be looking for them. In your approach the logs would show a much more noticeable error.  Whereas if the URL path changes, it's the reverse.